packages/webapp/pages/posts/[id]/index.tsx

```diff
@@ -154,16 +159,28 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
   ) : (
     <SquadPostPageNavigation squadLink={post.source.permalink} />
   );
-  const articleNavigation = router?.query?.squad ? (
-    <Link href={`/squads/${router.query.squad}`}>
-      <a className="flex flex-row items-center font-bold text-theme-label-tertiary typo-callout">
-        <ArrowIcon size={IconSize.Medium} className="mr-2 -rotate-90" />
-        Back to {router.query.n || 'Squad'}
-      </a>
-    </Link>
-  ) : (
-    <></>
-  );
+
+  const articleNavigation = (() => {
+    const routedFromSquad = router?.query?.squad;
+    const squadLink = `/squads/${router.query.squad}`;
+
+    if (layout === FeedLayout.V1) {
+      return <PostBackButton link={routedFromSquad ? squadLink : undefined} />;
+    }
+
+    if (!routedFromSquad) {
```

Comment:
Same here on the positive condition note, nothing to worry & non need to change, just want to highlight in case you think is better as well :)

Reply:
It is more of a defensive programming where I return early then I leave the most complicated result at the bottom of the function to make the early returns cleaner and easier to read.